lib/trie/database.go

```diff
@@ -348,6 +378,18 @@ func (t *Trie) writeDirty(db chaindb.Batch, curr Node) error {
 		}
 	}
 
+	l, ok := curr.(*node.Leaf)
+	if ok {
+		// see if there is a child trie with this hash
+		childTrieHash := common.NewHash(l.Value)
+		childTrie, childTrieexists := t.childTries[childTrieHash]
+		if childTrieexists {
+			if err := childTrie.writeDirty(db, childTrie.root); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
same here - only check subtrie with `ChildStorageKeyPrefix`

Reply:
alternatively can't you just iterate througn the `childTries` map and call `writeDirty` on them?

Reply:
addressed